rllib/models/experimental/torch/encoder.py

```diff
@@ -2,59 +2,60 @@
 import torch.nn as nn
 import tree
 
-from ray.rllib.models.experimental.base import (
-    ForwardOutputType,
-    ModelConfig,
-)
+from ray.rllib.models.experimental.base import ModelConfig, Model
 from ray.rllib.models.experimental.encoder import (
     Encoder,
     STATE_IN,
     STATE_OUT,
+    ENCODER_OUT,
+    ACTOR,
+    CRITIC,
 )
-from ray.rllib.models.temp_spec_classes import TensorDict
-from ray.rllib.policy.sample_batch import SampleBatch
-from ray.rllib.utils.annotations import override
-from ray.rllib.policy.rnn_sequencing import add_time_dimension
+from ray.rllib.models.experimental.torch.primitives import TorchMLP, TorchModel
 from ray.rllib.models.specs.specs_dict import SpecDict
-from ray.rllib.models.specs.checker import check_input_specs, check_output_specs
 from ray.rllib.models.specs.specs_torch import TorchTensorSpec
-from ray.rllib.models.experimental.torch.primitives import TorchMLP, TorchModel
-
-ENCODER_OUT: str = "encoder_out"
+from ray.rllib.policy.rnn_sequencing import add_time_dimension
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.utils.annotations import override
+from ray.rllib.utils.nested_dict import NestedDict
 
 
 class TorchMLPEncoder(TorchModel, Encoder):
-    """A fully connected encoder."""
-
     def __init__(self, config: ModelConfig) -> None:
         TorchModel.__init__(self, config)
         Encoder.__init__(self, config)
 
+        # Create the neural networks
         self.net = TorchMLP(
             input_dim=config.input_dim,
             hidden_layer_dims=config.hidden_layer_dims,
             output_dim=config.output_dim,
             hidden_layer_activation=config.hidden_layer_activation,
         )
 
-    @property
-    @override(TorchModel)
-    def input_spec(self) -> SpecDict:
-        return SpecDict(
-            {SampleBatch.OBS: TorchTensorSpec("b, h", h=self.config.input_dim)}
+        # Define the input spec
+        self.input_spec = SpecDict(
+            {
+                SampleBatch.OBS: TorchTensorSpec("b, h", h=self.config.input_dim),
+                STATE_IN: None,
+                SampleBatch.SEQ_LENS: None,
+            }
         )
 
-    @property
-    @override(TorchModel)
-    def output_spec(self) -> SpecDict:
-        return SpecDict(
-            {ENCODER_OUT: TorchTensorSpec("b, h", h=self.config.output_dim)}
+        # Define the output spec
+        self.output_spec = SpecDict(
+            {
+                ENCODER_OUT: TorchTensorSpec("b, h", h=self.config.output_dim),
+                STATE_OUT: None,
+            }
         )
 
-    @check_input_specs("input_spec", cache=False)
-    @check_output_specs("output_spec", cache=False)
-    def forward(self, inputs: TensorDict, **kwargs) -> ForwardOutputType:
-        return {ENCODER_OUT: self.net(inputs[SampleBatch.OBS])}
+    @override(Model)
+    def _forward(self, inputs: NestedDict, **kwargs) -> NestedDict:
```

Comment:
Where we checking for specs?

Reply:
In the forward() method and the __call__() method for TF.
I'm unifying torch/tf to _forward.
We can unify most of the models this way.